hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/container/ReplicationManager.java

```diff
@@ -746,8 +746,9 @@ private void handleOverReplicatedContainer(final ContainerInfo container,
         if (excess > 0) {
           sendDeleteCommand(container, r.getDatanodeDetails(), true);
           excess -= 1;
+        } else {
+          break;
         }
-        break;
```

Comment:
Looking into this PR change, I am thinking if we still need to do excess check before sending delete command for unhealthy containers. As we want to remove all unhealthy container replicas, we can just send delete command and decrease the excess count by the way.

I prefer simplified the logic like below:
```java
      for (ContainerReplica r : unhealthyReplicas) {
          sendDeleteCommand(container, r.getDatanodeDetails(), true);
          excess -= 1;
      }
```

Reply:
@linyiqun What if there is an excess of containers, but all containers are somehow unhealthy? We don't want RM to remove all copies of them, as that could result in a lost container. That is why we limit the number of unhealthy containers removed to the excess number.

Reply:
Okay, get it. Thanks @sodonnel.